core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -333,22 +333,25 @@ object Extraction {
 
     def newTuple(root: JValue, mappings: List[Mapping]): Any = {
       root match {
-        case JArray(items) if items.length >= 1 && items.length <= 22 =>
+        case JArray(items) if items.nonEmpty && items.length <= tuples.length =>
           val builtItems: Seq[Object] = items.zip(mappings).map({
-            case (i,m) =>
-              build(i, m).asInstanceOf[Object]
+            case (item, mapping) =>
+              build(item, mapping).asInstanceOf[Object]
           })
-          val tupleClass = Class.forName("scala.Tuple" + items.length)
+          val tupleIndex = items.length - 1
+          val tupleClass = tuples.lift(tupleIndex).getOrElse {
+            throw new IllegalArgumentException(s"Cannot instantiate a tuple of length ${items.length} even though that should be a valid tuple length.")
```

Comment:
Still think it may be worth turning `tuples` into a `Map` (only doing the `.getConstructors()(0)` bit once during the singleton intialization).

The error here also doesn't seem quite right (at this point we actually couldn't load the tuple class, so while technically we can't instantiate it there's a deeper reason). Additionally, if we hadn't been able to get this class, we would have blown up while initializing the whole singleton object. I'm wondering if we just do a `.get` here and point to the initialization of `tuples` to show that if we've gotten to this point in the code, we know that entry exists.